### PR TITLE
Prac 08 Feedback Request

### PR DESCRIPTION
**Pull request by nicknlowe**
@Kanos812 - would you be able to review my code and provide any feedback? It would be greatly appreciated.

Thanks!

### CODE REVIEW

File: prac_08/box_layout_demo.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from kivy.app import App
+from kivy.lang import Builder
+
+
+class BoxLayoutDemo(App):
+    def build(self):
+        self.title = "Box Layout Demo"
+        self.root = Builder.load_file('box_layout.kv')
```
Comment (Kanos812) [inline prac_08/box_layout_demo.py:8]: I'm being a bit nitpicky but I think it's important to have a consistent use of single or double quotes for strings

File: prac_08/box_layout.kv [hunk 2]
```diff
@@ -0,0 +1,16 @@
+BoxLayout:
```
Comment (Kanos812) [inline prac_08/box_layout.kv:1]: Good practice to include docstrings to provide code overview and provide other details (date of creation, author etc.)

File: prac_08/box_layout_demo.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    def handle_greet(self):
+        self.root.ids.output_label.text = f"Hello {self.root.ids.input_name.text}"
+
+    def press_clear(self):
+        self.root.ids.input_name.text = ""
+        self.root.ids.output_label.text = ""
+
+
+BoxLayoutDemo().run()
```
Comment (Kanos812) [inline prac_08/box_layout_demo.py:19]: The code itself looks fine and seems to output correctly, however an extension to improve the code would be adding error handling. Although its a bit 'extra' I think it's good practice

File: prac_08/convert_miles_km.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        self.root.ids.input_miles.text = str(value)
+        self.handle_calculation()
+
+    def handle_calculation(self):
+        """
+        Handle calculation and output result to output label widget.
+        :return: None
+        """
+        value = self.get_valid_miles()
+        result = value * MILES_TO_KILOMETRES
```
Comment (Kanos812) [inline prac_08/convert_miles_km.py:52]: Nitpicking again but should have consistent formatting (python has all lowercase with underscores)

File: prac_08/convert_miles_km.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        Handle up/down button press.
+        :param change: the amount to change
+        :return: None
+        """
+        value = self.get_valid_miles() + change
+        self.root.ids.input_miles.text = str(value)
+        self.handle_calculation()
+
+    def handle_calculation(self):
```
Comment (Kanos812) [inline prac_08/convert_miles_km.py:46]: Naming conventions could be improved to increase readability and decrease ambiguity. For example 'handle_calculation' could be called 'calculate_and_display_kilometres' which is far more descriptive

**Review by Kanos812 — COMMENTED** [review]
Overall the code is neat and looks to work as expected. I don't see any major issues with syntax or structure

There are just a couple of habits which I think are important to try and integrate into your programming:
- Be consistent with your formatting - for strings and naming conventions, you sometimes switch between different methods (for example some strings have double quotations while some have single) 
- Include more comments to provide greater content and clarity to the code both for an external reader and for yourself (for example if you come back after a long time you may have forgotten why a specific section was put there)
- Update your docstrings with your name and provide details about the file for better readability 
- In some instances a more descriptive and appropriate name would fit better in place of what already exists 

I am being very nitpicky (trust me I forget most of these myself) however they're the one per centers which will make programming just a little bit easier. Otherwise well done! Definitely made it easy for me to go through and read your code 

PS - One thing I also just noticed is a very low number of commits. I highly encourage you to practice using the commit button more regularly at important stages as this is very good practice

**Comment by nicknlowe** [comment]
> Overall the code is neat and looks to work as expected. I don't see any major issues with syntax or structure
> 
> There are just a couple of habits which I think are important to try and integrate into your programming:
> 
> * Be consistent with your formatting - for strings and naming conventions, you sometimes switch between different methods (for example some strings have double quotations while some have single)
> * Include more comments to provide greater content and clarity to the code both for an external reader and for yourself (for example if you come back after a long time you may have forgotten why a specific section was put there)
> * Update your docstrings with your name and provide details about the file for better readability
> * In some instances a more descriptive and appropriate name would fit better in place of what already exists
> 
> I am being very nitpicky (trust me I forget most of these myself) however they're the one per centers which will make programming just a little bit easier. Otherwise well done! Definitely made it easy for me to go through and read your code
> 
> PS - One thing I also just noticed is a very low number of commits. I highly encourage you to practice using the commit button more regularly at important stages as this is very good practice

Thank you for your great feedback! Will definitely keep it in mind for Assignment 2. I really appreciate it 😁

**Pull request merged**